### DISCUSSION TITLE
create unique QueryType object across files

### DISCUSSION THREAD

**Discussion by qlibp** [mirumee/ariadne#630] [Ideas]
create unique QueryType object across files,  so that we could separate our files more easily. Currently, we could separate the literal schema easily because of the nature of string. But when it comes to bindables, we are only able to pass the bindables around files so that it could associate correctly to the resolver functions. It would be nice if `make_executable_schema` could resolve this for us.

For now, I make a work around to get a unique QueryType bindables across files.

**Comment by rafalp** [comment]
I'm not sure I understand. Could you provide an example of what you are after?

**Reply by qlibp** [reply]
say that you call `QueryType()` in different files and then you would got two different objects, and when I pass these two bindable objects inside `make_executable_schema()`, I would then get error like "There can be only one type named 'Query' ". Therefore, for me, it seems to be a good practice to making `QueryType()` return the same object

**Reply by rafalp** [reply]
Oh, this is not a case with Ariadne. You can have multiple `QueryType` instances and you can pass them to `make_executable_schema`. This is supported and its one of ways to split code into smaller filers.

What you can't have is multiple `type Query` in schema. But you can solve this by having one `type Query` and multiple `extend type Query`.

**Reply by qlibp** [reply]
> Oh, this is not a case with Ariadne. You can have multiple `QueryType` instances and you can pass them to `make_executable_schema`. This is supported and its one of ways to split code into smaller filers.
> 
> What you can't have is multiple `type Query` in schema. But you can solve this by having one `type Query` and multiple `extend type Query`.

I realize this just now, thanks! And I also find out that, documentation inside the graphql definition with `extend` keyword will crash the parser.

```
schema_defs = '''
    """
    test documentation
    """
    extend type Query {
        helloTest: String!
    }
'''
```

**Reply by rafalp** [reply]
Looks like a suggestion for [GraphQL spec work group](https://github.com/graphql/graphql-spec).
